FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean

```diff
@@ -92,21 +108,39 @@ noncomputable
 instance BaseChange.algebra : Algebra (FiniteAdeleRing A K) (FiniteAdeleRing B L) :=
   RingHom.toAlgebra (FiniteAdeleRing.mapRingHom A K L B)
 
+attribute [local instance 9999] Algebra.toSMul in
+noncomputable instance baseChangeAlgebra : Algebra K (FiniteAdeleRing B L) :=
+  Algebra.compHom (FiniteAdeleRing B L) (algebraMap K L)
+
+instance : IsScalarTower K L (FiniteAdeleRing B L) :=
+  IsScalarTower.of_algebraMap_eq' rfl
+
+attribute [local instance 9999] Algebra.toSMul in
+instance : IsScalarTower K (FiniteAdeleRing A K) (FiniteAdeleRing B L) := by
+  apply IsScalarTower.of_algebraMap_eq
+  intro x
+  nth_rw 2 [RingHom.algebraMap_toAlgebra]
+  symm
+  exact SemialgHom.commutes (FiniteAdeleRing.mapSemialgHom A K L B) x
```

Comment:
You could have done `exact (SemialgHom.commutes (FiniteAdeleRing.mapSemialgHom A K L B) x).symm` but I'm not bothered :-)